db/database.go

```diff
@@ -286,13 +300,15 @@ func NewDatabaseContext(dbName string, bucket base.Bucket, autoImport bool, opti
 	var err error
 	dbContext.sequences, err = newSequenceAllocator(bucket, dbContext.DbStats.Database())
 	if err != nil {
-		return nil, err
+		returnedError = err
+		return nil, returnedError
 	}
 
```

Comment:
Do we want to be adding cleanup for all the other stuff too?

It looks like there's potentially a lot of cleanup required throughout - not sure how harmful it is to be left as-is, could alwas spin out into a new ticket and get this in just for the observed stats issue.

```suggestion

	cleanupFunctions = append(cleanupFunctions, func() {
		dbContext.sequences.Stop()
	})

```

Reply:
Made an attempt to do this now. Might as well given the PR is fairly contained